auth-foundation/src/main/java/com/okta/authfoundation/client/OidcConfiguration.kt

```diff
@@ -132,8 +132,18 @@ class OidcConfiguration private constructor(
 
     companion object {
         internal fun defaultJson(): Json = Json { ignoreUnknownKeys = true }
-
-        lateinit var default: OidcConfiguration
+        private var _default: OidcConfiguration? = null
+
+        /**
+         * The default OidcConfiguration. This must be set before calling any OAuth flows. Note that this variable
+         * can only be set once in the lifetime of the app.
+         */
+        var default: OidcConfiguration
+            get() = _default ?: throw IllegalStateException("Attempted to use OidcConfiguration.default without setting it")
+            set(value) {
+                if (_default == null) _default = value
+                else throw IllegalStateException("Attempted setting OidcConfiguration.default after initialization")
+            }
```

Comment:
Are we planning on building out a configuration file, similar to the Swift version, that can implicitly load the default configuration?

Reply:
I looked into this a bit, but I think I should handle this in a separate PR. There are a few different ways of achieving this, and I need to dig into this more to see what would be ideal. All the options I found so far have some downsides. Adding config file wouldn't be a breaking API change.